pkg/collector/corechecks/oracle/activity.go

```diff
@@ -0,0 +1,430 @@
+		}
+		if sample.LogonTime.Valid {
+			sessionRow.LogonTime = sample.LogonTime.String
+		}
+		if sample.ClientIdentifier.Valid {
+			sessionRow.ClientIdentifier = sample.ClientIdentifier.String
+		}
+		if sample.BlockingInstance != nil {
+			sessionRow.BlockingInstance = *sample.BlockingInstance
+		}
+		if sample.BlockingSession != nil {
+			sessionRow.BlockingSession = *sample.BlockingSession
+		}
+		if sample.FinalBlockingInstance != nil {
+			sessionRow.FinalBlockingInstance = *sample.FinalBlockingInstance
+		}
+		if sample.FinalBlockingSession != nil {
+			sessionRow.FinalBlockingSession = *sample.FinalBlockingSession
+		}
+		if sample.WaitEvent.Valid {
+			sessionRow.WaitEvent = sample.WaitEvent.String
+		}
+		if sample.WaitEventGroup.Valid {
+			sessionRow.WaitEventGroup = sample.WaitEventGroup.String
+		}
+		if sample.WaitTimeMicro != nil {
+			sessionRow.WaitTimeMicro = *sample.WaitTimeMicro
+		}
+
+		statement := ""
+		obfuscate := true
+		if sample.Statement.Valid && sample.Statement.String != "" && !previousSQL {
+			statement = sample.Statement.String
+		} else if previousSQL {
+			if sample.PrevSQLFullText.Valid && sample.PrevSQLFullText.String != "" {
+				statement = sample.PrevSQLFullText.String
+			} else {
+				/* The case where we got the previous sql but not the text. Happens when
+				 * a cursor with a short living explain plan was evicted from the shared pool.
+				 * We'll search for the text of a different cursor of the same SQL.
+				 */
+				err = c.db.Get(&statement, "SELECT sql_fulltext FROM v$sqlstats WHERE sql_id = :1 AND rownum=1", sqlPrevSQL.SQLID)
+				if err != nil {
+					log.Errorf("sql_text for the previous statement: %s", err)
+				}
+			}
+		} else if (sample.OpFlags & 8) == 8 {
+			statement = "LOG ON/LOG OFF"
+			obfuscate = false
+		} else if commandName != "" {
+			statement = commandName
+			//obfuscate = false
+		} else if sessionType == "BACKGROUND" {
+			statement = program
+			// The program name can contain an IP address
+			//obfuscate = false
+		} else if sample.Module.Valid && sample.Module.String == "DBMS_SCHEDULER" {
+			statement = sample.Module.String
+			obfuscate = false
+		} else {
+			log.Warnf("activity sql text empty for %#v \n", sample)
+		}
+		if statement != "" && obfuscate {
+			obfuscatedStatement, err := c.GetObfuscatedStatement(o, statement)
+			sessionRow.Statement = obfuscatedStatement.Statement
+			if err == nil {
+				sessionRow.Commands = obfuscatedStatement.Commands
+				sessionRow.Tables = obfuscatedStatement.Tables
+				sessionRow.Comments = obfuscatedStatement.Comments
+				sessionRow.QuerySignature = obfuscatedStatement.QuerySignature
+			}
+		} else {
+			sessionRow.Statement = statement
+			sessionRow.QuerySignature = common.GetQuerySignature(statement)
+		}
+
+		if sample.PdbName.Valid {
+			sessionRow.PdbName = sample.PdbName.String
+		}
+		sessionRow.CdbName = c.cdbName
+		sessionRows = append(sessionRows, sessionRow)
+	}
+	o.Stop()
```

Comment:
Would it make sense to use `defer` here?

Reply:
Added `defer`: https://github.com/DataDog/datadog-agent/blob/nenad.noveljic/dbm-oracle/pkg/collector/corechecks/oracle-dbm/activity.go#L402